src/components/okta_auth.js

```diff
@@ -0,0 +1,50 @@
+// Okta Auth Component for React
+// Check if a session exists and handle the response
+
+import React, { Component, PropTypes } from 'react';
+
+export default class OktaAuth extends Component {
+  static contextTypes = {
+    router: PropTypes.object,
+    location: PropTypes.object
+  };
+
+  componentWillMount() {
```

Comment:
componenWillMount runs at the first mount, while the others runs when you come back to that route, since the component don't mount again if you change route and come back